src/sentry/api/endpoints/event_ai_suggest.py

```diff
@@ -0,0 +1,217 @@
+    for idx, exc in enumerate(reversed((event.get("exception") or {}).get("values") or ())):
+        exception = {}
+        if idx > 0:
+            exception["raised_during_handling_of_previous_exception"] = True
+        exception["type"] = exc["type"]
+        exception["message"] = exc.get("value")
+        mechanism = exc.get("mechanism") or {}
+        exc_meta = mechanism.get("meta")
+        if exc_meta:
+            exception["meta"] = exc_meta
+        if mechanism.get("handled") is False:
+            exception["unhandled"] = True
+
+        frames = exc.get("stacktrace", {}).get("frames")
+        first_in_app = False
+        if frames:
+            stacktrace = []
+            # TODO: consider truncating the stack
+            for frame in reversed(frames):
+                stack_frame = OrderedDict()
+                stack_frame["func"] = frame.get("function")
+                stack_frame["module"] = frame.get("module")
+                stack_frame["file"] = frame.get("filename")
+                stack_frame["line"] = frame.get("lineno")
+                if frame.get("in_app"):
+                    stack_frame["in_app"] = True
+                if frame["in_app"]:
+                    if first_in_app:
+                        stack_frame["crashed_here"] = True
+                        first_in_app = False
+                    line = (frame.get("context_line") or "").strip()
+                    if line:
+                        stack_frame["code"] = line
+                stacktrace.append(stack_frame)
+            if stacktrace:
+                exception["stacktrace"] = stacktrace
+        exceptions.append(exception)
+
+    msg = event.get("message")
+    if msg:
+        data["message"] = msg
+
+    return data
+
+
+@region_silo_endpoint
+class EventAiSuggestEndpoint(ProjectEndpoint):
+    # go away
+    private = True
+    enforce_rate_limit = True
+    rate_limits = {
+        "GET": {
+            RateLimitCategory.IP: RateLimit(5, 1),
+            RateLimitCategory.USER: RateLimit(5, 1),
+            RateLimitCategory.ORGANIZATION: RateLimit(5, 1),
+        },
+    }
+
+    def get(self, request: Request, project, event_id) -> Response:
+        """
+        Makes AI make suggestions about an event
+        ````````````````````````````````````````
+
+        This endpoint returns a JSON response that provides helpful suggestions about how to
+        understand or resolve an event.
+        """
+        event = eventstore.get_event_by_id(project.id, event_id)
+        if event is None:
+            raise ResourceDoesNotExist
+
+        if not features.has("organizations:ai-suggest", project.organization, actor=request.user):
+            raise ResourceDoesNotExist
+
+        # Cache the suggestion for a certain amount by primary hash, so even when new events
+        # come into the group, we are sharing the same response.
+        cache_key = "ai:" + event.get_primary_hash()
```

Comment:
Should we just use the `group_id` instead?

Does it also make sense to restrict this only to error issues?

Reply:
Unclear yet how this works with issue platform issues, want to actually discuss this with you. From what I can tell right now `get_event_by_id` from the event store returns an event with `group_id`, but our docs on that also say that we consider this deprecated which is why I did not want to build this into the cache.

I assume we're asking for the group ID as additional information, but I do not believe that the API actually needs that info for anything other than the cache key.

Reply:
Yeah, we need to clean this up a bit. Once performance issues have moved over to the issue platform, I can clean up a bunch of cruft here and make things clearer - at the moment we're in this split world where events can still have multiple groups associated with them, which is going away once perf issues are migrated. Once that's done, it'll be easier to clean up the rest of the interface here as well.